### PR TITLE
#2120 vaccine status

### PR DESCRIPTION
**Pull request by iamleeg**
I've made the schema changes and added the fields to the data downloads, but not made any changes to the line list or curator form. This at least allows us to quickly get the data in where ADI sources contain it, and we can decide later how the UI needs to support the fields.

### CODE REVIEW

File: data-serving/data-service/src/model/vaccine.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    },
+    { _id: false },
+);
+
+export type VaccineDocument = mongoose.Document & {
+    name: string;
+    batch: string;
+    date: Date;
+    sideEffects: [SymptomsDocument];
+    previousInfection: 'yes' | 'no' | 'NA';
```
Comment (abhidg) [inline data-serving/data-service/src/model/vaccine.ts:21]: Could this be boolean? That way presence of previousInfection would have true / false and absence indicates no previous infection or data not known.

**Review by abhidg — COMMENTED** [review]
Looks good - one field could be boolean. We should come up with a way to address #2103, but that can be done later.

**Comment by jim-sheldon** [comment]
IMHO the case fields workflow could get consolidated/simplified--
We have the`export` script which pulls field names from the database into a text file, now a `prepare` script to turn the text file into a JSON file, which then gets consumed by `controllers/case`.
Could we simply pull the fields from the database, then feed them into the case controller constructor?

File: data-serving/data-service/src/model/vaccine.ts (hunk 1 above)
Comment (iamleeg) [reply data-serving/data-service/src/model/vaccine.ts:21]: The request at #2120 was for yes/no/NA. I guess we have to ask @mougk and other experts what the expectation is.

**Review by jim-sheldon — APPROVED** [review]

**Comment by iamleeg** [comment]
@jim-sheldon Oh is the export script the canonical source of fields.txt (from the database)? I didn't know that! I don't know how that will work with this change: there's a list of vaccine documents for each case document, but we specifically want the CSV export to have space for exactly three vaccines per case.

**Comment by iamleeg** [comment]
The database shouldn't ever be the canonical source of what fields get included in the export, because we have fields that are present like `restrictedNotes` that should not be exported.

**Comment by jim-sheldon** [comment]
It looks like it will work just fine once the schema is applied.
Agreed the database shouldn't be the unqualified source...I'd like there to exist a `hidden` or `private` field in the [jsonSchema spec](https://docs.mongodb.com/v4.2/reference/operator/query/jsonSchema/#json-schema) but no dice...in the absence of that, we do `delete doc.restrictedNotes;` and supporting tests for that requirement exist.

**Review by abhidg — APPROVED** [review]

**Pull request merged**